### PR TITLE
Adds eslint to frontend

### PR DESCRIPTION
**Pull request by rdonati**
STILL NEED TO ADD TO BACKEND..
Not working when I add a config file to the backend for some reason.. someone else feel free to give this a shot

Adds eslint to frontend, uses standard defaults (NOT Airbnb which a lot of people supposedly prefer)
Adds vscode config to autoformat on save
Applies formatting to existing files in the frontend

### CODE REVIEW

**Review by Danny123234 — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
+  "eslint.validate": ["javascript"]
```
Comment (CosineP) [inline .vscode/settings.json:5]: i think we (also?) want typescript, typescriptreact

File: frontend/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    ecmaVersion: 12,
+    sourceType: "module"
+  },
+  plugins: [
+    "react",
+    "@typescript-eslint"
+  ],
+  rules: {
+    quotes: ["error", "double"],
```
Comment (CosineP) [inline frontend/.eslintrc.js:23]: Mentioned above. I'm advocating for single quotes not arbitrarily.

File: frontend/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    // Keeping no-unused-vars on had some weird behavior with enums
+    semi: ["error", "always"],
+    "no-unused-vars": "off",
```
Comment (CosineP) [inline frontend/.eslintrc.js:26]: Unused variables is imo one of the most useful lints out there. What issues with enums did we see? I think we have some enums which we never construct / read one value of, which actually *does* indicate a problem. Can we consider fixing, `// eslint-disable-next-line`-ing them, or at the very least, turning these to warnings?

File: frontend/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    // note you must disable the base rule as it can report incorrect errors
+    "no-use-before-define": "off",
+    "react/react-in-jsx-scope": "off",
+    // allow jsx syntax in js files (for next.js project)
+    "react/jsx-filename-extension": [1, { extensions: [".js", ".jsx", ".ts", ".tsx"] }] // should add ".ts" if typescript project
```
Comment (CosineP) [inline frontend/.eslintrc.js:31]: Who said anything about next.js? The advantage of keeping jsx to jsx/tsx is that .ts can use <> syntax, plus, that way the linter will ensure our file extensions are as-expected.

**Review by CosineP — CHANGES_REQUESTED** [review]
A couple of notes. Please read this discussion I had with @swjeong0825. https://github.com/Free-Laborers/u-refer/pull/37#discussion_r736134593 . In general, I proposed a three-step process, because this is going to absolutely suck with merges. As a result of that, we should also do this for frontend and backend *together* to avoiding needing to get to a good merge point twice.

First, get everyone on the team onboard with eslint (does everyone use vscode? Do they all have eslint installed? (no)). This needs to happen first so we don't get thrashing between formatting styles.

Second, someone (you, clearly) should do a full pass of auto-formatting (like what you've done here, with caveats) *at a time where we have as few unmerged lines of code as possible*. You should do it synchronously, with everyone on board that there's few unmerged lines, and then merge as quickly as possible, and get everyone to pull, to avoid as much headache as possible.

Third, eslint validation can and should go in CI.

One other point I want to make: We should choose a style that most closely matches what we already have. In particular, I think most of the edits I see are single quotes to double quotes. I think we mostly use single quotes as-is. So we should simply keep doing that. Then, I am very pro semicolons, but if we find there are extremely few, relative to needed, I think it would be worth enforcing no semicolons as opposed to the opposite.

**Pull request closed without merging**